cirq/value/random_state_test.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_parse_random_state():
+    global_state = np.random.get_state()
+
+    def rand(prng):
+        np.random.set_state(global_state)
+        return prng.rand()
+
+    prngs = [
+        np.random,
+        cirq.value.parse_random_state(np.random),
+        cirq.value.parse_random_state(None)
+    ]
+    vals = [rand(prng) for prng in prngs]
+    eq = cirq.testing.EqualsTester()
+    eq.add_equality_group(*vals)
+
+    seed = np.random.randint(-2**31, 2**31)
```

Comment:
Negative numbers aren't valid inputs to np.random.RandomState`:

```
>>> np.random.RandomState(-1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 644, in mtrand.RandomState.__init__
  File "mtrand.pyx", line 680, in mtrand.RandomState.seed
ValueError: Seed must be between 0 and 2**32 - 1
```

Reply:
Fixed